core/src/main/java/org/lflang/federated/generator/FedGenerator.java

```diff
@@ -466,6 +487,94 @@ private void analyzeFederates(Reactor federation, LFGeneratorContext context) {
       }
       federatesByInstantiation.put(instantiation, federateInstances);
     }
+
+    // Create the connections between federates.
+    // Each connection in the AST may represent more than one connection between
+    // federation instances because of banks and multiports. We need to generate communication
+    // for each of these. To do this, we create a ReactorInstance so that we don't have
+    // to duplicate the rather complicated logic in that class. We specify a depth of 1,
+    // so it only creates the reactors immediately within the top level, not reactors
+    // that those contain.
+    ReactorInstance mainInstance = new ReactorInstance(federation, messageReporter);
+
+    new ReactionInstanceGraph(mainInstance); // Constructor has side effects; its result is ignored
+
+    for (ReactorInstance child : mainInstance.children) {
+      for (PortInstance output : child.outputs) {
+        for (SendRange srcRange : output.getDependentPorts()) {
+          for (RuntimeRange<PortInstance> dstRange : srcRange.destinations) {
+            MixedRadixInt srcID = srcRange.startMR();
+            MixedRadixInt dstID = dstRange.startMR();
+            int dstCount = 0;
+            int srcCount = 0;
+
+            while (dstCount++ < dstRange.width) {
+              // FIXME: If connection is interleaved, these digits will be reversed?
```

Comment:
We should either address this or attach it to an issue.

Reply:
Added a test verifying correctness and removed comment in commit #3d6de027d.